scio-core/src/main/scala/com/spotify/scio/util/ScioUtil.scala

```diff
@@ -116,10 +116,11 @@ private[scio] object ScioUtil {
   }
 
   def tempDirOrDefault(tempDirectory: String, sc: ScioContext): ResourceId = {
-    Option(tempDirectory).map(toResourceId).getOrElse {
-      val tempLocationOpt: String = sc.options.getTempLocation
-      FileSystems.matchNewResource(tempLocationOpt, true)
-    }
+    Option(tempDirectory)
+      .orElse(Option(sc.options.getTempLocation))
+      .orElse(Try(sc.optionsAs[GcpOptions]).toOption.flatMap(x => Option(x.getGcpTempLocation)))
+      .map(toResourceId)
+      .getOrElse(throw new IllegalStateException("Unable to find any temp location"))
```

Comment:
Apologies missed this on the first go. `IllegalArgumentException` is what we use elsewhere, also updated the message to reflect what gets thrown in `getTempFile`:
```suggestion
      .getOrElse(throw new IllegalArgumentException("No temporary location was specified. Specify a temporary location via --tempLocation or PipelineOptions.setTempLocation."))
```